### ISSUE TITLE
TensorRT 7.1.3 issue on Jetson/TX2

### ISSUE THREAD

**Issue by w3sip** [NVIDIA/TensorRT#1062]
## Environment

**TensorRT Version**: 7.1.3
**NVIDIA GPU**: Jetson (TX2)
**NVIDIA Driver Version**:
**CUDA Version**: 10.2
**CUDNN Version**: 8.0
**Operating System**: Jetpack 4.4
**Python Version (if applicable)**: 
**Tensorflow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 
**Baremetal or Container (if so, version)**: baremetal

## Relevant Files

We have a TensorRT wrapper code, that has been used for a couple of years. We are seeing an issue executing inferences with most models with version 7.1.3: but only on TX2.

So, using, for example models_VGGNet_VOC0712_SSD_300x300.tar.gz mentioned [here](https://docs.nvidia.com/deeplearning/tensorrt/release-notes/tensorrt-5.html#rel_5-0-2)
- expected results are seen with 7.1.3 (CUDA 10.2 or 11) on Windows
- expected results are seen with 7.1.3 (CUDA 10.2) on x86_64 Linux
- expected results are seen with 6.0.1.5/CUDA 10.0 on Jetpack
- 7.1.3/CUDA 10.2 on Jetpack produces consistent (e.g. the same every time), but incorrect and different from all the other platforms, inference results

We're seeing this with most of our proprietary models as well, however there is at least one where it doesn't happen (e.g. results are consistent across all permutations).

In probing the individual layers' outputs when using that VGGNet model above, it seems like the first divergence between the 6.0.1.5 (and the rest of configurations) and 7.1.3 comes in "fc6" layer.

Setting BuilderFlag::kDEBUG doesn't make a difference. Not sure what else to try.

**Comment by hassansadiq1** [comment]
I have encountered same problem on jetson nano and xavier.
### Environment
**TensorRT Version: 7.1.3
CUDA Version: 10.2
CUDNN Version: 8.0
Operating System: Jetpack 4.5.1**
Our model was working fine with TensorRt 6 on Jetson family. But when I upgraded to new OS with TensorRt 7, it shows incorrect results. The problem comes with Jetson Family only.

I have tested it on my system with GTX 1080 Ti with same stack and it works fine.
**Ubuntu 18.04 
TensorRT Version: 7.1.3
CUDA Version: 10.2
CUDNN Version: 8.0**

**Comment by ttyio** [comment]
Hello @w3sip  @hassansadiq1 

We have verified model: ssd.prototxt, VGG_VOC0712_SSD_300x300_iter_120000.caffemodel
data: data/samples/ssd/bus.ppm

below is the result
![image](https://user-images.githubusercontent.com/365590/115004806-c9765780-9ed9-11eb-9b59-95b10b18aa5d.png)

No obvious difference observed, could you provide more info? thanks!

**Comment by w3sip** [comment]
That is either because 
a) Your test was building the engine without serializing/deserializing it, or
b) 7.1.3 on TX2 is not really 7.1.3: it has been patched without modifying the version number. 

My original issue was related to engine serialization bug in TensorRT plugins, and had been resolved in the context of this thread: https://github.com/NVIDIA/TensorRT/issues/1089

**Comment by ttyio** [comment]
@w3sip , 
The result come from QA, what about the accuracy issue you reported in this thread? How can we repro it?

**Comment by w3sip** [comment]
The issue ended up being not related to accuracy, but to engine corruption during serialization/deserialization. To reproduce, your test must serialize and de-serialize the engine before use, then used de-serialized object for inference.

**Comment by w3sip** [comment]
But, to reiterate -- this is fixed in 7.2.0, and has been backported to 7.1.3 in context of TensorRT-OSS. Please read the entire thread I've linked for details.

**Comment by ttyio** [comment]
Sorry @w3sip for the misunderstanding I did not read whole threads;(
So the mismatch because the plugin serialize and deserialized back wrong content in 7.1.3, and now it is fixed by backporting 7.2.0 to 7.1.3. 

Can I close these 2 bugs now? thanks!

**Comment by ttyio** [comment]
I will close this, please reopen if you still have question, thanks!
